python/ray/tune/examples/dcgan_mnist_pbt/dcgan_mnist_pbt.py

```diff
@@ -0,0 +1,364 @@
+class PytorchTrainable(tune.Trainable):
+    def _setup(self, config):
+        self.netD = Discriminator().to(device)
+        self.netD.apply(weights_init)
+        self.netG = Generator().to(device)
+        self.netG.apply(weights_init)
+        self.criterion = nn.BCELoss()
+        self.optimizerD = optim.Adam(self.netD.parameters(),
+                                     lr=config.get("lr", 0.01),
+                                     betas=(beta1, 0.999))
+        self.optimizerG = optim.Adam(self.netG.parameters(),
+                                     lr=config.get("lr", 0.01),
+                                     betas=(beta1, 0.999))
+        self.dataloader = get_data_loader()
+
+    def _train(self):
+        lossG, lossD, is_score = train(self.netD, self.netG, self.optimizerG,
+                                       self.optimizerD, self.criterion,
+                                       self.dataloader, self._iteration)
+        return {"lossg": lossG, "lossd": lossD, "is_score": is_score}
+
+    def _save(self, checkpoint_dir):
+        netD_path = os.path.join(checkpoint_dir, "netDmodel.pth")
+        torch.save(self.netD.state_dict(), netD_path)
+
+        netG_path = os.path.join(checkpoint_dir, "netGmodel.pth")
+        torch.save(self.netG.state_dict(), netG_path)
+        return checkpoint_dir
+
+    def _restore(self, checkpoint_dir):
+        self.netD.load_state_dict(
+            torch.load(os.path.join(checkpoint_dir, "netDmodel.pth")))
+        self.netG.load_state_dict(
+            torch.load(os.path.join(checkpoint_dir, "netGmodel.pth")))
+
+    def reset_config(self, new_config):
+        del self.optimizerD
+        del self.optimizerG
+        self.optimizerD = optim.Adam(self.netD.parameters(),
+                                     lr=new_config.get("netD_lr"),
+                                     betas=(beta1, 0.999))
+        self.optimizerG = optim.Adam(self.netG.parameters(),
+                                     lr=new_config.get("netG_lr"),
+                                     betas=(beta1, 0.999))
+        self.config = new_config
+        return True
+# __Trainable_end__
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--smoke-test", action="store_true", help="Finish quickly for testing")
+    args, _ = parser.parse_known_args()
+    ray.init()
+
+    dataloader = get_data_loader()
+    # Plot some training images
+    real_batch = next(iter(dataloader))
+    plt.figure(figsize=(8, 8))
+    plt.axis("off")
+    plt.title("Original Images")
+    plt.imshow(np.transpose(vutils.make_grid(
+        real_batch[0].to(device)[:64], padding=2, normalize=True).cpu(), (1, 2, 0)))
+
+    if not args.smoke_test:
+        plt.show()
+
+    # load the pretrained mnist classification model for inception_score
+    mnist_cnn = Net()
+    mnist_cnn.load_state_dict(torch.load("./mnist_cnn.pt"))
+    mnist_cnn.eval()
+    mnist_model_ref = ray.put(mnist_cnn)
```

Comment:
Instead of this, I think you can just specify a `restore=` parameter in `tune.run`

Reply:
oops nvm, misread